tests/test_bookmarks_metafields.py

```diff
@@ -0,0 +1,221 @@
+        That the bookmark is the maximum value sent to the target for the replication key.
+        That a second sync respects the bookmark
+            All data of the second sync is > the bookmark from the first sync
+            The number of records in the 2nd sync is less then the first (This assumes that
+                new data added to the stream is done at a rate slow enough that you haven't
+                doubled the amount of data from the start date to the first sync between
+                the first sync and second sync run in this test)
+
+        Verify that only data for incremental streams is sent to the target
+
+        PREREQUISITE
+        For EACH stream that is incrementally replicated there are multiple rows of data with
+            different values for the replication key
+        """
+        conn_id = self.create_connection(original_credentials=True)
+
+        # Select all streams and no fields within streams
+        found_catalogs = menagerie.get_catalogs(conn_id)
+        incremental_stream = {'metafields'}
+
+        # Our test data sets for Shopify do not have any abandoned_checkouts
+        our_catalogs = [catalog for catalog in found_catalogs if
+                        catalog.get('tap_stream_id') in incremental_stream]
+        self.select_all_streams_and_fields(conn_id, our_catalogs, select_all_fields=True)
+
+        # Run a sync job using orchestrator
+        first_sync_record_count = self.run_sync(conn_id)
+
+        # verify that the sync only sent records to the target for selected streams (catalogs)
+        self.assertEqual(set(first_sync_record_count.keys()),
+                         incremental_stream)
+
+        first_sync_state = menagerie.get_state(conn_id)
+
+        # Get data about actual rows synced
+        first_sync_records = runner.get_records_from_target_output()
+
+        # Run a second sync job using orchestrator
+        second_sync_record_count = self.run_sync(conn_id)
+
+        # Get data about rows synced
+        second_sync_records = runner.get_records_from_target_output()
+
+        first_max_bookmarks, first_min_bookmarks = self.min_max_bookmarks_by_stream(first_sync_records)
+        #first_sync_bookmarks = menagerie.get_state(conn_id)
+        
+        #######################
+        # Update State between Syncs
+        #######################
+
+        # new_state = {'bookmarks': dict()}
+        # simulated_states = self.calculated_states_by_stream(first_sync_bookmarks)
+
+        # for stream, updated_state in simulated_states.items():
+        #     new_state['bookmarks'][stream] = updated_state
+        # menagerie.set_state(conn_id, new_state)
+
+        _ , second_min_bookmarks = self.min_max_bookmarks_by_stream(second_sync_records)
+
+        for stream in incremental_stream:
+            with self.subTest(stream=stream):
+
+                # verify that you get less data the 2nd time around
```

Comment:
First letter should be camel-case. Please make this change in other comments as well.

```suggestion
                # Verify that you get less data the 2nd time around
```